spec/meta_inspector/redirections_spec.rb

```diff
@@ -47,6 +47,25 @@
         expect(page.url).to eq("http://blogs.clarionledger.com/dechols/2014/03/24/digital-medicine/?nclick_check=1")
       end
     end
+
+    context "when there is a callback to be ran between redirects that blocks redirections to private IP addresses" do
+      it "raises an exception" do
+        stub_request(:get, "https://www.facebook.com/")
+          .to_return(:status => 302,
+                     :headers => { "Location" => "http://10.0.0.0/" })
+
+        redirect_options = {
+          callback: proc do |_previous_response, next_request|
+            ip_address = Resolv.getaddress(next_request.url.host)
+            raise StandardError if IPAddr.new(ip_address).private?
```

Comment:
Can we raise a custom error instead? As this is using a generic `StandardError`, it took me a while to realize that I didn't have the `Resolv` gem installed, until I saw this other `StandardError`:

```ruby
Failure/Error: ip_address = Resolv.getaddress(next_request.url.host)
     
     NameError:
       uninitialized constant Resolv
     
                   ip_address = Resolv.getaddress(next_request.url.host)
                                ^^^^^^
```